core/templates/domain/skill/skill-backend-api.service.spec.ts

```diff
@@ -103,159 +94,213 @@ describe('Skill backend API service', function() {
+    fakeAsync( () => {
+      const successHandler = jasmine.createSpy('success');
+      const failHandler = jasmine.createSpy('fail');
+      let skillDict = null;
+      skillBackendApiService.fetchSkill('1').then(
+        data => skillDict = data.skill);
 
-    $httpBackend.expect('PUT', '/skill_editor_handler/data/1').respond(
-      500, 'Error on update skill 1.');
-    SkillBackendApiService.updateSkill(
-      skillDict.id, skillDict.version, 'commit message', []
-    ).then(successHandler, failHandler);
-    $httpBackend.flush();
+      const req = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
 
-    expect(successHandler).not.toHaveBeenCalled();
-    expect(failHandler).toHaveBeenCalledWith('Error on update skill 1.');
-  });
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(sampleResponse);
+      flushMicrotasks();
+
+      skillBackendApiService.updateSkill(
+        skillDict.id, skillDict.version, 'commit message', []
+      ).then(successHandler, failHandler);
+
+      const req2 = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
+
+      expect(req2.request.method).toEqual('PUT');
+
+      req2.flush('Error on update skill 1.', {
+        status: 500,
+        statusText: 'Error on update skill 1.'
+      });
+      flushMicrotasks();
+
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalledWith('Error on update skill 1.');
+    }
+    ));
 
   it('should succesfully fetch multiple existing skills from the backend',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync( () => {
+      const successHandler = jasmine.createSpy('success');
+      const failHandler = jasmine.createSpy('fail');
 
-      var skillDataUrl = '/skill_data_handler/' + encodeURIComponent('1,2');
-      $httpBackend.expect('GET', skillDataUrl).respond(sampleResponse2);
-      SkillBackendApiService.fetchMultiSkills(['1', '2']).then(
+      skillBackendApiService.fetchMultiSkills(['1', '2']).then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      const skillDataUrl = '/skill_data_handler/' + encodeURIComponent('1,2');
+      const req = httpTestingController.expectOne(skillDataUrl);
+
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(sampleResponse2);
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith(sampleResponse2.skills);
       expect(failHandler).not.toHaveBeenCalled();
-    });
+    }
+    ));
 
-  it('should use the rejection handler if fetch multiple skills from the ' +
-    'backend failed', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should use the rejection handler if fetch multiple skills from the' +
+  +'backend failed',
```

Comment:
I think you're missing a space and have an extra '+' here?

Reply:
Done!